test/shmemx/team_max.c

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2019 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <shmem.h>
+#include <shmemx.h>
+
+#define NUM_TEAMS 64
+
+int ret = 0;
+int dest_ret = 0;
+
+int main(void)
+{
+    int me, npes, i;
+    int team_count = 0;
+    int team_size=0;
+
+    shmem_init();
+
+    me = shmem_my_pe();
+    npes = shmem_n_pes();
+
+    shmemx_team_t new_team[NUM_TEAMS];
+
+    i = 0;
+    while(i < 9) {
+        
+        team_size = i+1;
+        team_size = team_size % npes;
+        if(team_size == 0){
+            i++;
+            continue;
+        }
```

Comment:
This conditional looks like it will cause problems.  The split, sync, and reduce operations below are collective and must be called by all PEs in `SHMEMX_TEAM_WORLD` but some PEs will end the iteration here.

Reply:
This iteration will be stopped for all PE. I added this condition for a situation when (i+1)==npes. Because in that case (i+1)%npes will give 0. Hence all the PEs will come across same situation for this iteration.

Reply:
Interesting, I'm not sure we specified this case fully.